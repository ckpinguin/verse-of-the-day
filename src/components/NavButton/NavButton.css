.NavButton {
    cursor: pointer;
    background: Transparent;
    border: 0.5px solid #ccc;
    border-radius: 5px;    
    margin: auto;
    padding: 7px 7px 7px 7px;
}