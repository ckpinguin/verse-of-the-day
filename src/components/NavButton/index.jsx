import React from 'react';
import PropTypes from 'prop-types';

import './index.css';

import { isDebug, debug } from '../../debug';

NavButton.propTypes = {
    icon: PropTypes.string,
    onClick: PropTypes.func.isRequired
};
NavButton.defaultProps = {
    icon: 'fa fa-question'
};
function NavButton({ icon, onClick }) {
    return (
        <button
            className={`NavButton ${icon}`}
            style={ isDebug ? debug.borderStyle : {} }
            onClick={onClick}
        />
    );
}
export default NavButton;