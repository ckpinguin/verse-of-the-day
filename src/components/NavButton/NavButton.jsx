import React from 'react';
import PropTypes from 'prop-types';

import './NavButton.css';

import { isDebug, debug } from '../../debug';

NavButton.propTypes = {
    icon:       PropTypes.string,
    id:         PropTypes.string,
    onClick:    PropTypes.func.isRequired
};
NavButton.defaultProps = {
    icon: 'fa fa-question'
};
function NavButton({ id, icon, onClick }) {
    return (
        <button
            id={id}
            className={`NavButton ${icon}`}
            style={ isDebug ? debug.borderStyle : {} }
            onClick={onClick}
        />
    );
}
export default NavButton;