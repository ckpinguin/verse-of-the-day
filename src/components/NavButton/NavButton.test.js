import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme'; 
import renderer from 'react-test-renderer';

import NavButton from '.';

describe('<NavButton />'    , () => {
    let props;
    let mountedNavButton;
    let shallowNavButton;
    const onClick = jest.fn();    
    const navButton = () => {
        if (!mountedNavButton) {
            mountedNavButton = mount(
                <NavButton {...props} />
            );
        }
        return mountedNavButton;
    };
    const navButtonShallow = () => {
        if (!shallowNavButton) {
            shallowNavButton = shallow(
                <NavButton {...props} />
            );
        }
        return shallowNavButton;
    };
        
    beforeEach(() => {
        // App = require('./App').default;        
        props = {
            icon: 'fa fa-question',
            onClick: onClick
        };
        mountedNavButton = undefined;
        shallowNavButton = undefined;
    });
    
    it('renders without crashing', () => {
        ReactTestUtils.renderIntoDocument(<NavButton {...props} />);
    });

    it('renders shallow', () => {
        navButtonShallow();
    });

    it('mounts successfully (full DOM rendering)', () => {
        navButton();
    });

    it('allows us to set props', () => {
        const wrapper = navButton();
        expect(wrapper.props().icon).toEqual('fa fa-question');
        wrapper.setProps({ icon: 'bonka' });
        expect(wrapper.props().icon).toEqual('bonka');
    });

    it('always renders exactly one div', () => {
        const divs = navButtonShallow().find('div'); // shallow goes only 1 level deep
        expect(divs.length).toBe(1);
    });

    it('outputs the correct className for the div', () => {
        const div = navButtonShallow().find('div');
        expect(div.node.props.className).toBe('NavButton fa fa-question');
    });

    it('did not change UI compared to recorded snapshot', () => {
        const component = renderer.create(<NavButton {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});