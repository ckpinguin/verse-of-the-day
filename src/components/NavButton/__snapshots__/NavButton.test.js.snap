// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButton /> did not change UI compared to recorded snapshot 1`] = `
<div
  className="NavButton fa fa-question"
  id={undefined}
  onClick={[Function]}
  style={Object {}}
/>
`;
